xclim/sdba/_processing.py

```diff
@@ -158,17 +158,47 @@ def _reordering(ds, *, dim):
     def _reordering_1d(data, ordr):
         return np.sort(data)[np.argsort(np.argsort(ordr))]
 
-    return (
-        xr.apply_ufunc(
-            _reordering_1d,
-            ds.sim,
-            ds.ref,
-            input_core_dims=[[dim], [dim]],
-            output_core_dims=[[dim]],
-            vectorize=True,
-            dask="parallelized",
-            output_dtypes=[ds.sim.dtype],
+    def _reordering_2d(data, ordr):
+        data_r = data.ravel()
+        ordr_r = ordr.ravel()
+        reorder = np.sort(data_r)[np.argsort(np.argsort(ordr_r))]
+        return reorder.reshape(data.shape)[
+            :, int(data.shape[0] / 2)
```

Comment:
This should `data.shape[1]` no? `window` is the second dimension below. 

Also, I think that `map_blocks` does this automatically for you, depending on how you call it. I think `reordered=[Grouper.DIM]` would already do this, I'm not 100% sure. @aulemahal ?

Reply:
oui oups en effet pour [1].

not sure how to do that ? it fails if I don't return something the size of time.

Reply:
Indeed, `map_groups` should slice to the middle of the window if the results still has a `window` dimension. That would require the `apply_ufunc` to declare `window` in its `output_core_dims`. However, as the result is the same, I think that slicing directly in the ufunc make more sense, and might yield better performances!